Sources/PenguinStructures/FixedSizeArray.swift

```diff
@@ -0,0 +1,269 @@
+//******************************************************************************
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// The point of this prototype is to prove that we can generate efficient code
+// for single-element insertion and deletion on a statically-sized array, with
+// stable types for the end products i.e.,
+// 
+//   type(of: a.removing(at: i).inserting(x, at: j)) == type(of: a)
+//
+// and
+//
+//    type(of: a.inserting(x, at: j).removing(at: i)) == type(of: a)
+
+
+/// Statically-sized nonempty collections of homogeneous elements.
+///
+/// This protocol should be thought of as an implementation detail of `ArrayN`;
+/// it is not generally useful.
+public protocol FixedSizeArray : MutableCollection, RandomAccessCollection,
+                          CustomStringConvertible where Index == Int
```

Comment:
I'm sorry, I'm not quite sure I follow here; could you explain why we need `FixedSizeArray` to refine `CustomStringConvertible`?

Reply:
We could conform `Array0` and `ArrayN` explicitly and separately if you like, but it doesn't seem like a win to me.

Reply:
Ah, okay, SGTM.